pyvrp/read.py

```diff
@@ -102,13 +103,31 @@ def read(
     # assume duration == distance.
     durations = distances = round_func(instance["edge_weight"])
 
-    dimension: int = instance.get("dimension", durations.shape[0])
-    depot_idcs: np.ndarray = instance.get("depot", np.array([0]))
-    num_vehicles: int = instance.get("vehicles", dimension - 1)
+    dimension = instance.get("dimension", durations.shape[0])
+    depot_idcs = instance.get("depot", np.array([0]))
+    num_vehicles = instance.get("vehicles", dimension - 1)
 
-    capacity: int = instance.get("capacity", _INT_MAX)
-    if capacity != _INT_MAX:
-        capacity = round_func(np.array([capacity])).item()
+    if isinstance(instance.get("capacity"), Number):
+        capacities = round_func(np.full(num_vehicles, instance["capacity"]))
+    elif instance.get("capacity") is not None:
+        capacities = round_func(instance["capacity"])
+    else:
+        capacities = np.full(num_vehicles, _INT_MAX)
+
+    if "vehicles_depot" in instance:
+        vehicles_depots = instance["vehicles_depot"] - 1  # make 0-indexed
+    else:
+        vehicles_depots = np.full(num_vehicles, depot_idcs[0])
+
+    if "vehicles_fixed_cost" in instance:
+        fixed_costs = round_func(instance["vehicles_fixed_cost"])
+    else:
+        fixed_costs = np.zeros(num_vehicles, dtype=np.int64)
+
+    if "vehicles_variable_cost" in instance:
+        variable_costs = round_func(instance["vehicles_variable_cost"])
```

Comment:
The cost is going to get really big (x1000 variable cost and x1000 distance), but I don't see a way around it. The variable costs are given in two decimal places.

Reply:
Yeah that's a bit annoying but agree, I don't really see a way around this.